plugins/woocommerce/client/blocks/assets/js/blocks/mini-cart/iapi-frontend.ts

```diff
@@ -215,6 +239,62 @@ store< MiniCart >(
 			},
 		},
 
+		actions: {
+			openDrawer() {
+				if ( onCartClickBehaviour === 'navigate_to_checkout' ) {
+					window.location.href = checkoutUrl;
+					return;
+				}
+				state.isOpen = true;
+			},
+
+			closeDrawer() {
+				state.isOpen = false;
+				state.miniCartButtonRef?.focus();
+			},
+
+			overlayCloseDrawer( e: MouseEvent ) {
+				// Only close the drawer if the overlay itself was clicked.
+				if ( e.target === e.currentTarget ) {
+					miniCartActions.closeDrawer();
+				}
+			},
+
+			handleOverlayKeydown: withSyncEvent( ( e: KeyboardEvent ) => {
+				if ( state.isOpen ) {
+					if ( e.key === 'Escape' ) {
+						miniCartActions.closeDrawer();
+					}
+
+					// Trap focus if it is an overlay (main menu).
+					if ( e.key === 'Tab' ) {
+						const { ref } = getElement();
+						const focusableElements = getFocusableElements( ref );
```

Comment:
Do we have to do it on every keydown? Can we store/cache the list somewhere?

Reply:
> Do we have to do it on every keydown? Can we store/cache the list somewhere?

I guess we could try to do it. The problem I see is that I am not sure when this "cache" should be refreshed. As the minicart is interactive, focusable elements might change. For example, when adding or removing items from the cart. We could refresh the cache when the items change, but maybe there are extensions or blocks doing other things that could change the focusable elements (For example, adding a dropdown in the minicart). I thought doing it per keydown was safer and `querySelectorAll` shouldn't be too costly, but I'd love to know other opinions.